chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -214,170 +75,11 @@
-    group "root"
-    mode "0644"
-    source "crowbar_join.logrotate.suse"
-    action :create
-  end
-
-  # remove old crowbar_join.sh file
-  file "/etc/init.d/crowbar_join.sh" do
-    action :delete
-  end
-
-  if node["platform"] == "suse"
-    ## make sure the repos are properly setup
-    repos = Provisioner::Repositories.get_repos(provisioner_server_node, node["platform"], node["platform_version"])
-    for name, attrs in repos
-      url = %x{zypper --non-interactive repos #{name} 2> /dev/null | grep "^URI " | cut -d : -f 2-}
-      url.strip!
-      if url != attrs[:url]
-        unless url.empty?
-          Chef::Log.info("Removing #{name} zypper repository pointing to wrong URI...")
-          %x{zypper --non-interactive removerepo #{name}}
-        end
-        Chef::Log.info("Adding #{name} zypper repository...")
-        %x{zypper --non-interactive addrepo --refresh #{attrs[:url]} #{name}}
-      end
-    end
-    # install additional packages
-    os = "#{node[:platform]}-#{node[:platform_version]}"
-    if node[:provisioner][:packages][os]
-      node[:provisioner][:packages][os].each { |p| package p }
-    end
-  end
+  include_recipe "provisioner::client"
 end
 
-aliaz = begin
-  display_alias = node["crowbar"]["display"]["alias"]
-  if display_alias && !display_alias.empty?
-    display_alias
-  else
-    node["hostname"]
-  end
-rescue
-  node["hostname"]
-end
-
-%w(/etc/profile.d/zzz-prompt.sh /etc/profile.d/zzz-prompt.csh).each do |cfg|
-  template cfg do
-    source "zzz-prompt.sh.erb"
-    owner "root"
-    group "root"
-    mode "0644"
-
-    variables(
-      prompt_from_template: proc { |user, cwd|
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", user) \
-          .gsub("CWD", cwd) \
-          .gsub("SUFFIX", "${prompt_suffix}") \
-          .gsub("ALIAS", aliaz) \
-          .gsub("HOST", node["hostname"]) \
-          .gsub("FQDN", node["fqdn"])
-      },
-
-      zsh_prompt_from_template: proc {
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", "%{\\e[0;31m%}%n%{\\e[0m%}") \
-          .gsub("CWD", "%{\\e[0;35m%}%~%{\\e[0m%}") \
-          .gsub("SUFFIX", "%#") \
-          .gsub("ALIAS", "%{\\e[0;35m%}#{aliaz}%{\\e[0m%}") \
-          .gsub("HOST", "%{\\e[0;35m%}#{node["hostname"]}%{\\e[0m%}") \
-          .gsub("FQDN", "%{\\e[0;35m%}#{node["fqdn"]}%{\\e[0m%}")
-      },
-
-      bash_prompt_from_template: proc {
-        node["provisioner"]["shell_prompt"].to_s \
-          .gsub("USER", "\\[\\e[01;31m\\]\\u\\[\\e[0m\\]") \
-          .gsub("CWD", "\\[\\e[01;31m\\]\\w\\[\\e[0m\\]") \
-          .gsub("SUFFIX", "${prompt_suffix}") \
-          .gsub("ALIAS", "\\[\\e[01;35m\\]#{aliaz}\\[\\e[0m\\]") \
-          .gsub("HOST", "\\[\\e[01;35m\\]#{node["hostname"]}\\[\\e[0m\\]") \
-          .gsub("FQDN", "\\[\\e[01;35m\\]#{node["fqdn"]}\\[\\e[0m\\]")
-      }
-    )
-  end
-end
+include_recipe "provisioner::shell-prompt"
+include_recipe "provisioner::shell-vars"
```

Comment:
I thought it could be useful in other places and I didn't see any harm in defining them everywhere.  But I have no problem dropping them from non-admin nodes if you prefer.